reconcile/openshift_base.py

```diff
@@ -556,6 +559,82 @@ def check_unused_resource_types(ri):
             logging.warning(msg)
 
 
+def _normalize_secret(secret: OR):
+    body = secret.body
+    string_data = body.pop("stringData", None)
+    if string_data:
+        body.setdefault("data", {})
+        for k, v in string_data.items():
+            v = base64.b64encode(str(v).encode()).decode("utf-8")
+            body["data"][k] = v
+
+
+def _normalize_noop(item: OR):
+    pass
+
+
+def normalize_object(item: OR, deep_copy=True) -> OR:
+    if deep_copy:
+        n = copy.deepcopy(item)
+    else:
+        n = item
+
+    n.body["metadata"].setdefault("annotations", {})
+    n.remove_qontract_annotations()
+
+    # Remove K8s managed attributes not neede to compare objects
+    n.body["metadata"].pop("creationTimestamp", None)
+    n.body["metadata"].pop("resourceVersion", None)
+    n.body["metadata"].pop("generation", None)
+    n.body["metadata"].pop("selfLink", None)
+    n.body["metadata"].pop("uid", None)
+    n.body["metadata"].pop("fieldRef", None)
+    n.body["metadata"].pop("managedFields", None)
+    n.body["metadata"].pop("namespace", None)
```

Comment:
To avoid duplicate logic, can extract a `K8S_MANAGED_ATTRIBUTES` then return a new dict:

```python
    K8S_MANAGED_ATTRIBUTES = {"creationTimestamp", "resourceVersion", "generation", ...}
    {k: v for k, v in n.body["metadata"].items() if k not in K8S_MANAGED_ATTRIBUTES}
```

Reply:
done

Reply:
small typo `neede` -> `needed`

Reply:
fixed.